### ISSUE TITLE
@SuppressWarnings annotation does not suppress UncommentedMain

### ISSUE THREAD

**Issue by MEZk** [checkstyle/checkstyle#2646]
**Problem:** @SuppressWarnings annotation does not suppress UncommentedMainCheck violations.

**Input file: `Test1.java`**

```java
public class Test1 {
    /**
     * Javadoc
     */
    @SuppressWarnings("checkstyle:uncommentedmain")
    public static void main(String[] args) {
    }
}
```

**Input file: `Test2.java`**

```java
public class Test2 {
    /**
     * Javadoc
     */
    @SuppressWarnings("checkstyle:UncommentedMain")
    public static void main(String[] args) {
    }
}
```

**Input file: `Test3.java`**

```java
public class Test3 {
    /**
     * Javadoc
     */
    @SuppressWarnings("uncommentedmain")
    public static void main(String[] args) {
    }
}
```

**Input file: `Test4.java`**

```java
public class Test4 {
    /**
     * Javadoc
     */
    @SuppressWarnings("UncommentedMain")
    public static void main(String[] args) {
    }
}
```

**Configuration file: `my_check.xml`**
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name = "Checker">
    <property name="charset" value="UTF-8"/>
    <property name="severity" value="warning"/>
   <module name="SuppressWarningsFilter" />
    <module name="TreeWalker">
       <module name="SuppressWarningsHolder" />
       <module name="UncommentedMain"/>
    </module>
</module>
```

**Checkstyle 6.12.2 output**
````
andreiselkin@andreiselkin ~/Downloads/temp $ java -jar checkstyle-6.12.1-all.jar -c my_check.xml Test1.java
Starting audit...
/home/andreiselkin/Downloads/temp/Test1.java:5: warning: Uncommented main method found.
Audit done.

andreiselkin@andreiselkin ~/Downloads/temp $ java -jar checkstyle-6.12.1-all.jar -c my_check.xml Test2.java
Starting audit...
/home/andreiselkin/Downloads/temp/Test2.java:5: warning: Uncommented main method found.
Audit done.

andreiselkin@andreiselkin ~/Downloads/temp $ java -jar checkstyle-6.12.1-all.jar -c my_check.xml Test3.java
Starting audit...
/home/andreiselkin/Downloads/temp/Test3.java:5: warning: Uncommented main method found.
Audit done.

andreiselkin@andreiselkin ~/Downloads/temp $ java -jar checkstyle-6.12.1-all.jar -c my_check.xml Test4.java
Starting audit...
/home/andreiselkin/Downloads/temp/Test4.java:5: warning: Uncommented main method found.
Audit done.
````

The following configuration also does not help:
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name = "Checker">
    <property name="charset" value="UTF-8"/>
    <property name="severity" value="warning"/>
    <module name="SuppressWarningsFilter" />
    <module name="TreeWalker">
       <module name="SuppressWarningsHolder" />
       <module name="UncommentedMain">
         <property name="id" value="checkstyle:uncommentedmain"/>
      </module>
    </module>
</module>
```

**Comment by romani** [comment]
might be related issue : #2250 .
